src/functions/errorMessageGenerator.ts

```diff
@@ -12,15 +13,15 @@ export function errorMessageGenerator(
     if (linkStartsWithHttps(url)) {
       if (isLinkValid(url)) {
         if (isLinkDuplicate(url, items)) {
-          return "This one I've got already!";
+          return errorMessages.find((t) => t.state == "duplicate")!.message;
```

Comment:
- Prefer `===` over `==`.
- What if the predicate in `.find(...)` method does not find a value? Then we're accessing a `.message` property on undefined which is going to fail.
- [Optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) might be helpful in such cases.

Reply:
- Yeah, == is a mistake.
- Good point. Great.